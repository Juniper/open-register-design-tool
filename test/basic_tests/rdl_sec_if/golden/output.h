//   Ordt 230719.01 autogenerated file 
//   Input: ./rdl_sec_if/test.rdl
//   Parms: ./rdl_sec_if/test.parms
//   Date: Thu Jul 20 13:46:50 EDT 2023
//

#ifndef __BASE_MAP_REGISTER_MAP__
#define __BASE_MAP_REGISTER_MAP__

#include <bits.h>

/* BASE_MAP_REGISTERS memory map */
enum BASE_MAP_REGS {
  BASE_REGS_CONFIG_REGS_0 = 0x0,
  BASE_REGS_CONFIG_REGS_1 = 0x4,
  BASE_REGS_CONFIG_REGS_2 = 0x8,
  BASE_REGS_CONFIG_REGS_3 = 0xc,
  BASE_REGS_CONFIG_REGS_4 = 0x10,
  BASE_REGS_CONFIG_REGS_5 = 0x14,
  BASE_REGS_CONFIG_REGS_6 = 0x18,
  BASE_REGS_CONFIG_REGS_7 = 0x1c,
  PRIMARY_BASE_REGS_CONFIG_REGS_0 = 0x20,
  PRIMARY_BASE_REGS_CONFIG_REGS_1 = 0x24,
  PRIMARY_BASE_REGS_CONFIG_REGS_2 = 0x28,
  PRIMARY_BASE_REGS_CONFIG_REGS_3 = 0x2c,
  PRIMARY_BASE_REGS_CONFIG_REGS_4 = 0x30,
  PRIMARY_BASE_REGS_CONFIG_REGS_5 = 0x34,
  PRIMARY_BASE_REGS_CONFIG_REGS_6 = 0x38,
  PRIMARY_BASE_REGS_CONFIG_REGS_7 = 0x3c
};

/* BASE_REGS_CONFIG_REGS_0 (Config reg set  config_regs_0) register fields */
#define BASE_REGS_CONFIG_REGS_0_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define BASE_REGS_CONFIG_REGS_0_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* BASE_REGS_CONFIG_REGS_1 (Config reg set  config_regs_1) register fields */
#define BASE_REGS_CONFIG_REGS_1_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define BASE_REGS_CONFIG_REGS_1_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* BASE_REGS_CONFIG_REGS_2 (Config reg set  config_regs_2) register fields */
#define BASE_REGS_CONFIG_REGS_2_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define BASE_REGS_CONFIG_REGS_2_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* BASE_REGS_CONFIG_REGS_3 (Config reg set  config_regs_3) register fields */
#define BASE_REGS_CONFIG_REGS_3_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define BASE_REGS_CONFIG_REGS_3_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* BASE_REGS_CONFIG_REGS_4 (Config reg set  config_regs_4) register fields */
#define BASE_REGS_CONFIG_REGS_4_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define BASE_REGS_CONFIG_REGS_4_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* BASE_REGS_CONFIG_REGS_5 (Config reg set  config_regs_5) register fields */
#define BASE_REGS_CONFIG_REGS_5_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define BASE_REGS_CONFIG_REGS_5_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* BASE_REGS_CONFIG_REGS_6 (Config reg set  config_regs_6) register fields */
#define BASE_REGS_CONFIG_REGS_6_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define BASE_REGS_CONFIG_REGS_6_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* BASE_REGS_CONFIG_REGS_7 (Config reg set  config_regs_7) register fields */
#define BASE_REGS_CONFIG_REGS_7_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define BASE_REGS_CONFIG_REGS_7_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* PRIMARY_BASE_REGS_CONFIG_REGS_0 (Config reg set  config_regs_0) register fields */
#define PRIMARY_BASE_REGS_CONFIG_REGS_0_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define PRIMARY_BASE_REGS_CONFIG_REGS_0_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* PRIMARY_BASE_REGS_CONFIG_REGS_1 (Config reg set  config_regs_1) register fields */
#define PRIMARY_BASE_REGS_CONFIG_REGS_1_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define PRIMARY_BASE_REGS_CONFIG_REGS_1_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* PRIMARY_BASE_REGS_CONFIG_REGS_2 (Config reg set  config_regs_2) register fields */
#define PRIMARY_BASE_REGS_CONFIG_REGS_2_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define PRIMARY_BASE_REGS_CONFIG_REGS_2_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* PRIMARY_BASE_REGS_CONFIG_REGS_3 (Config reg set  config_regs_3) register fields */
#define PRIMARY_BASE_REGS_CONFIG_REGS_3_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define PRIMARY_BASE_REGS_CONFIG_REGS_3_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* PRIMARY_BASE_REGS_CONFIG_REGS_4 (Config reg set  config_regs_4) register fields */
#define PRIMARY_BASE_REGS_CONFIG_REGS_4_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define PRIMARY_BASE_REGS_CONFIG_REGS_4_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* PRIMARY_BASE_REGS_CONFIG_REGS_5 (Config reg set  config_regs_5) register fields */
#define PRIMARY_BASE_REGS_CONFIG_REGS_5_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define PRIMARY_BASE_REGS_CONFIG_REGS_5_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* PRIMARY_BASE_REGS_CONFIG_REGS_6 (Config reg set  config_regs_6) register fields */
#define PRIMARY_BASE_REGS_CONFIG_REGS_6_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define PRIMARY_BASE_REGS_CONFIG_REGS_6_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

/* PRIMARY_BASE_REGS_CONFIG_REGS_7 (Config reg set  config_regs_7) register fields */
#define PRIMARY_BASE_REGS_CONFIG_REGS_7_LSB_FIELD GENMASK(15, 0)    /* Bunch o lsb bits */
#define PRIMARY_BASE_REGS_CONFIG_REGS_7_MSB_FIELD GENMASK(31, 16)    /* Bunch o msb bits */

#endif
