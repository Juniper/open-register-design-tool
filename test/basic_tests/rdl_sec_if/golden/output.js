/*   Ordt 190725.01 autogenerated file 
 *   Input: ./rdl_sec_if/test.rdl
 *   Parms: ./rdl_sec_if/test.parms
 *   Date: Sun Nov 17 17:01:02 EST 2019
 */

register_set base_map "L1 regs" {

  address = 0x0;
  register_width = 32;
  register_set base_regs "base_regs register_set" {
    address = 0x0;
    
    register config_regs "Config reg set  config_regs" {
      address = 0x0;
      category = { DYNAMIC_CONFIG };
      repeat = 8;
      integer msb_field[16] "Bunch o msb bits" param {
        access_mode = READ_WRITE;
        reset = 0x0;
      };
      integer lsb_field[16] "Bunch o lsb bits" param {
        access_mode = READ_WRITE;
        reset = 0xffff;
      };
    };
    
    register_set_size = 0x20;
  };
  
  register_set primary "L2 regs" {
    address = 0x20;
    
    register_set base_regs "base_regs register_set" {
      address = 0x0;
      
      register config_regs "Config reg set  config_regs" {
        address = 0x0;
        category = { DYNAMIC_CONFIG };
        repeat = 8;
        integer msb_field[16] "Bunch o msb bits" param {
          access_mode = READ_WRITE;
          reset = 0x0;
        };
        integer lsb_field[16] "Bunch o lsb bits" param {
          access_mode = READ_WRITE;
          reset = 0xffff;
        };
      };
      
      register_set_size = 0x20;
    };
    
    register_set_size = 0x20;
  };
  
  register_set_size = 0x40;
};

