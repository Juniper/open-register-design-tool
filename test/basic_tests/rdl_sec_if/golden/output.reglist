//   Ordt 190725.01 autogenerated file 
//   Input: ./rdl_sec_if/test.rdl
//   Parms: ./rdl_sec_if/test.parms
//   Date: Sun Nov 17 17:02:54 EST 2019
//

0x0              size: 32   fields: 2    bits: 32        base_regs.config_regs_0
0x4              size: 32   fields: 2    bits: 32        base_regs.config_regs_1
0x8              size: 32   fields: 2    bits: 32        base_regs.config_regs_2
0xc              size: 32   fields: 2    bits: 32        base_regs.config_regs_3
0x10             size: 32   fields: 2    bits: 32        base_regs.config_regs_4
0x14             size: 32   fields: 2    bits: 32        base_regs.config_regs_5
0x18             size: 32   fields: 2    bits: 32        base_regs.config_regs_6
0x1c             size: 32   fields: 2    bits: 32        base_regs.config_regs_7
0x20             size: 32   fields: 2    bits: 32        primary.base_regs.config_regs_0
0x24             size: 32   fields: 2    bits: 32        primary.base_regs.config_regs_1
0x28             size: 32   fields: 2    bits: 32        primary.base_regs.config_regs_2
0x2c             size: 32   fields: 2    bits: 32        primary.base_regs.config_regs_3
0x30             size: 32   fields: 2    bits: 32        primary.base_regs.config_regs_4
0x34             size: 32   fields: 2    bits: 32        primary.base_regs.config_regs_5
0x38             size: 32   fields: 2    bits: 32        primary.base_regs.config_regs_6
0x3c             size: 32   fields: 2    bits: 32        primary.base_regs.config_regs_7
