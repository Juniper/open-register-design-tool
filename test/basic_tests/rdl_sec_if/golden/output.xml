<?xml version="1.0" encoding="UTF-8"?>
<!--   Ordt 190725.01 autogenerated file 
       Input: ./rdl_sec_if/test.rdl
       Parms: ./rdl_sec_if/test.parms
       Date: Sun Nov 17 17:08:23 EST 2019
 -->

<map version="190725.01">
  <id>base_map</id>
  <baseaddr>0x0</baseaddr>
  <shorttext>L1 regs</shorttext>
  <regset>
    <id>base_regs</id>
    <shorttext>base_regs registers</shorttext>
    <baseaddr>0x0</baseaddr>
    <reg>
      <id>config_regs</id>
      <parentpath>base_regs</parentpath>
      <shorttext>Config reg set  config_regs</shorttext>
      <baseaddr>0x0</baseaddr>
      <width>32</width>
      <reps>8</reps>
      <stride>0x4</stride>
      <field>
        <id>lsb_field</id>
        <shorttext>Bunch o lsb bits</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>ffff</reset>
        <lowidx>0</lowidx>
        <width>16</width>
      </field>
      <field>
        <id>msb_field</id>
        <shorttext>Bunch o msb bits</shorttext>
        <access>RW</access>
        <hwinfo>
          <hwaccess>r</hwaccess>
        </hwinfo>
        <reset>0</reset>
        <lowidx>16</lowidx>
        <width>16</width>
      </field>
      <catcode>2</catcode>
      <access>RW</access>
    </reg>
    <highaddr>0x1f</highaddr>
  </regset>
  <map>
    <id>primary</id>
    <shorttext>L2 regs</shorttext>
    <baseaddr>0x20</baseaddr>
    <extroot>1</extroot>
    <regset>
      <id>base_regs</id>
      <parentpath>primary</parentpath>
      <shorttext>base_regs registers</shorttext>
      <baseaddr>0x20</baseaddr>
      <reg>
        <id>config_regs</id>
        <parentpath>primary.base_regs</parentpath>
        <shorttext>Config reg set  config_regs</shorttext>
        <baseaddr>0x20</baseaddr>
        <width>32</width>
        <reps>8</reps>
        <stride>0x4</stride>
        <field>
          <id>lsb_field</id>
          <shorttext>Bunch o lsb bits</shorttext>
          <access>RW</access>
          <hwinfo>
            <hwaccess>r</hwaccess>
          </hwinfo>
          <reset>ffff</reset>
          <lowidx>0</lowidx>
          <width>16</width>
        </field>
        <field>
          <id>msb_field</id>
          <shorttext>Bunch o msb bits</shorttext>
          <access>RW</access>
          <hwinfo>
            <hwaccess>r</hwaccess>
          </hwinfo>
          <reset>0</reset>
          <lowidx>16</lowidx>
          <width>16</width>
        </field>
        <catcode>2</catcode>
        <access>RW</access>
      </reg>
      <highaddr>0x3f</highaddr>
    </regset>
    <highaddr>0x3f</highaddr>
  </map>
</map>
